src/renderer/views/pool/PoolDetailsView.tsx

```diff
@@ -72,38 +75,62 @@ export const PoolDetailsView: React.FC = () => {
     }
   }, [oRouteAsset, setSelectedPoolAsset])
 
-  const routeAssetRD = RD.fromOption(oRouteAsset, () =>
-    Error(intl.formatMessage({ id: 'routes.invalid.asset' }, { asset }))
-  )
-
   const priceSymbol = useObservableState(selectedPricePoolAssetSymbol$, O.none)
 
   const priceRatio = useObservableState(priceRatio$, ONE_BN)
 
+  const isHistoryLoading = useObservableState(isHistoryLoading$, false)
+
   const poolDetailRD: PoolDetailRD = useObservableState(selectedPoolDetail$, RD.initial)
 
-  return FP.pipe(
-    RD.combine(routeAssetRD, poolDetailRD),
-    RD.fold(
-      renderInitialView,
-      renderPendingView,
-      (e: Error) => {
-        return <PoolStatus label={intl.formatMessage({ id: 'common.error' })} displayValue={e.message} />
-      },
-      ([asset, poolDetail]) => {
-        return (
-          <PoolDetails
-            asset={O.some(asset)}
-            depth={getDepth(poolDetail, priceRatio)}
-            volume24hr={get24hrVolume(poolDetail, priceRatio)}
-            allTimeVolume={getAllTimeVolume(poolDetail, priceRatio)}
-            totalSwaps={0 /* getTotalSwaps(history) */}
-            totalStakers={0 /* getTotalStakers(poolDetail) */}
-            priceUSD={getPriceUSD(poolDetail, priceRatio)}
-            priceSymbol={O.toUndefined(priceSymbol)}
-          />
+  const onRefreshData = useCallback(() => {
+    reloadSelectedPoolDetail()
+    reloadActionsHistory()
+  }, [reloadSelectedPoolDetail, reloadActionsHistory])
+
+  const refreshButtonDisabled = useMemo(() => {
+    return isHistoryLoading || FP.pipe(poolDetailRD, RD.isPending)
+  }, [isHistoryLoading, poolDetailRD])
+
+  const prevProps = useRef<TargetPoolDetailProps>(defaultDetailsProps)
+
+  return (
+    <>
+      <Styled.ControlsContainer>
+        <Styled.BackLink />
+        <RefreshButton clickHandler={onRefreshData} disabled={refreshButtonDisabled} />
+      </Styled.ControlsContainer>
+      {FP.pipe(
+        oRouteAsset,
+        O.fold(
+          () => <ErrorView title={intl.formatMessage({ id: 'routes.invalid.asset' }, { asset })} />,
+          (asset) =>
+            FP.pipe(
+              poolDetailRD,
+              RD.fold(
+                () => <PoolDetails asset={asset} {...defaultDetailsProps} />,
+                () => <PoolDetails asset={asset} {...prevProps.current} isLoading />,
+                ({ message }: Error) => {
+                  return <ErrorView title={message} />
+                },
+                (poolDetail) => {
+                  prevProps.current = {
+                    depth: getDepth(poolDetail, priceRatio),
+                    volume24hr: get24hrVolume(poolDetail, priceRatio),
+                    allTimeVolume: getAllTimeVolume(poolDetail, priceRatio),
+                    totalSwaps: 0 /* getTotalSwaps(history) */,
+                    totalStakers: 0 /* getTotalStakers(poolDetail) */,
+                    priceUSD: getPriceUSD(poolDetail, priceRatio),
```

Comment:
Similar to [previous message](https://github.com/thorchain/asgardex-electron/pull/1249#discussion_r605742418): Just push `poolDetail` into `PoolDetail` and do the rest to get `depth`, `volume24hr`, `allTimeVolume` etc, there.

Reply:
yep. will be changed with another PR to avoid conflicts resolving (as discussed in DMs)